src/Compilers/CSharp/Test/Semantic/SourceGeneration/GeneratorDriverTests_Attributes_FullyQualifiedName.cs

```diff
@@ -1233,6 +1236,69 @@ class YAttribute : System.Attribute { }
         Assert.Equal(1, counter);
     }
 
+    [Fact, WorkItem(66451, "https://github.com/dotnet/roslyn/issues/66451")]
+    public void MultipleInputs_RemoveFirst_ModifySecond()
+    {
+        var source0 = "public class GenerateAttribute : System.Attribute { }";
+        var comp0 = CreateCompilation(source0).VerifyDiagnostics().EmitToImageReference();
+
+        var generator = new IncrementalGeneratorWrapper(new PipelineCallbackGenerator(ctx =>
+        {
+            var provider = ctx.ForAttributeWithMetadataName<ClassDeclarationSyntax>("GenerateAttribute");
+            ctx.RegisterSourceOutput(provider, static (spc, syntax) => spc.AddSource(
+                $"{syntax.Identifier.Text}.g",
+                $"partial class {syntax.Identifier.Text} {{ /* generated */ }}"));
+        }));
+
+        var parseOptions = TestOptions.RegularPreview;
+
+        var source1 = """
+            [Generate]
+            [System.Obsolete]
+            public partial class Class1 { }
+            """;
+        var source2 = """
+            [Generate]
+            [System.Obsolete]
+            public partial class Class2 { }
+            """;
+
+        Compilation compilation = CreateCompilation(new[] { source1, source2 }, new[] { comp0 }, options: TestOptions.DebugDllThrowing, parseOptions: parseOptions);
+
+        GeneratorDriver driver = CSharpGeneratorDriver.Create(new[] { generator }, parseOptions: parseOptions);
+        verify(ref driver, compilation);
+
+        // Remove Class1 from the final provider via a TransformNode
+        // (by removing the Generate attribute).
+        replace(ref compilation, parseOptions, "Class1", """
+            //[Generate]
+            [System.Obsolete]
+            public partial class Class1 { }
+            """);
+        verify(ref driver, compilation);
+
+        // Modify Class2 (make it internal).
+        replace(ref compilation, parseOptions, "Class2", """
+            [Generate]
+            [System.Obsolete]
+            internal partial class Class2 { }
+            """);
+        verify(ref driver, compilation);
```

Comment:
nit: If I understood correctly, the main thing we're verifying is that we didn't assert/crash. Consider also verifying the correct output from the incremental generator at each step.